docs/directory-structure.md

```diff
@@ -0,0 +1,114 @@
+Directory structure
+===================
+
+This page documents the Zulip directory structure and how to decide
+where to put a file.
+
+Scripts
+-------
+
+* `scripts/` Scripts that production deployments might run manually
+  (e.g., `restart-server`).
+
+* `scripts/lib/` Scripts that are needed on production deployments but
+  humans should never run.
+
+* `scripts/setup/` Tools that production deployments will only run
+  once, during installation.
+
+* `tools/` Development tools.
+---------------------------------------------------------
+
+Bots
+----
+
+* `api/integrations/` Bots distributed as part of the Zulip API bundle.
+
+* `bots/` Previously Zulip internal bots. These usually need a bit of
+   work.  
+-----------------------------------------------------
```

Comment:
In markdown, you need a newline after a bulleted list before you can put other text, otherwise you'll get things misformatted like this:

![image](https://cloud.githubusercontent.com/assets/2746074/15682331/ffb09d6e-2711-11e6-9fda-a248320b4eb1.png)